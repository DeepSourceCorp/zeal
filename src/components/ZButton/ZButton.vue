<template>
  <component
    :is="type === 'link' || to ? 'a' : 'button'"
    :href="to"
    class="z-btn"
    v-bind="$attrs"
    :disabled="disabled"
    v-on="$listeners"
    :class="[
      `${color && `z-btn--${color}`}`,
      `${size && `z-btn--${size}`}`,
      {
        'is-disabled': this.disabled !== false,
        'is-active': this.active !== false,
        'full-width': this.fullWidth !== false
      },
      customClasses,
      `${!color && !customClasses ? 'z-btn--primary' : ''}`
    ]"
    :type="type"
  >
    <slot>Click</slot>
  </component>
</template>

<script>
export default {
  name: 'ZButton',
  props: {
    color: {
      default: '',
      type: String
    },
    type: {
      default: 'button',
      type: String
    },
    customClasses: {
      default: '',
      type: String
    },
    size: {
      default: '',
      type: String
    },
    disabled: {
      default: false,
      type: Boolean
    },
    active: {
      default: false,
      type: Boolean
    },
    fullWidth: {
      default: false,
      type: Boolean
    },
    to: {
      default: '',
      type: String
    }
  },
  methods: {
    handleClick(event) {
      this.$emit('click', event)
    }
  }
}
</script>

<style lang="css" scoped>
/* Common Button Style */

.z-btn {
  @apply inline-flex items-center px-6 text-base font-medium border-transparent h-10 rounded-sm relative;
}

.is-disabled {
  @apply opacity-50 cursor-not-allowed;
}

.full-width {
  @apply w-full inline-block;
}

/* Button - Primary Style */

.z-btn--primary {
  @apply bg-juniper hover:bg-light_juniper text-ink-400;
}

.z-btn--primary.is-disabled {
  @apply hover:bg-juniper;
}

/* Button - Secondary Style */

.z-btn--secondary {
  @apply bg-ink-300 hover:bg-ink-200 text-vanilla-100 border border-slate;
}

.z-btn--secondary.is-disabled {
  @apply hover:bg-ink-300;
}

.z-btn--secondary.is-active {
  @apply bg-ink-100;
}

.z-btn--link {
  @apply font-normal text-juniper hover:underline;
}

.z-btn--link.is-active {
  @apply underline;
}

.z-btn--link.is-disabled {
  @apply text-slate no-underline;
}

/* Different Sizes of the Button*/
.z-btn--small {
  @apply h-8 text-xs;
}

.z-btn--medium {
  @apply h-10 text-base;
}

.z-btn--large {
  @apply h-12 text-lg;
}

.z-btn--xlarge {
  @apply h-16 text-lg;
}
</style>
