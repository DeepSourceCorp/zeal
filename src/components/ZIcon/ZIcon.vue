<template>
    <svg 
      :class="[
        'z-icon',
        `${size && `z-icon--${size}`}`,
        `${icon && `z-icon--${icon}`}`
      ]"
      viewBox="0 0 24 24" 
      v-html="getIcon(icon)"></svg>
</template>

<script lang="ts">
import { Component, Prop, Vue } from 'vue-property-decorator'
import feather from 'feather-icons'
import customIcons from '../../utils/icon-helper'

@Component({
  name: 'ZIcon'
})
export default class Icon extends Vue {
  @Prop({ default: '' })
  private icon!: string
  @Prop({default: ''})
  private size!: string

  public getIcon(iconName: string): HTMLOrSVGImageElement {
    let DOM = null
    if (feather.icons[iconName]) {
      DOM = feather.icons[iconName] && feather.icons[iconName].contents
    } else {
      DOM = customIcons[iconName] &&  customIcons[iconName].contents;
    }
    return DOM
  }
}
</script>

<style lang="css" scoped>
  .z-icon {
    @apply  w-6 h-6 fill-current stroke-current text-slate stroke-2;
  }

  .z-icon--small {
    @apply w-4 h-4;
  }
</style>
