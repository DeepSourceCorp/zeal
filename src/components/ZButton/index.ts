import ZButton from './ZButton.vue';

export { ZButton };
