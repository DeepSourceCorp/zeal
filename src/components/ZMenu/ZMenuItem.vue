<template>
  <div
    class="z-menu-item flex items-center gap-2 px-3 py-3 hover:bg-ink-100 leading-none outline-none focus:outline-none"
  >
    <z-icon v-if="icon" :icon="icon" size="small" color="vanilla-100"></z-icon>
    <slot></slot>
  </div>
</template>

<script lang="ts">
import ZIcon from '../ZIcon/ZIcon.vue'

export default {
  name: 'ZMenuItem',
  components: {
    ZIcon
  },
  props: {
    icon: {
      type: String
    },
    disabled: {
      type: Boolean,
      default: false
    },
    action: {
      type: Function
    }
  }
}
</script>
