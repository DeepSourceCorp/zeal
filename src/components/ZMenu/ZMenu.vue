<template>
  <span class="relative z-menu">
    <button v-on:click="toggle" class="outline-none focus:outline-none" ref="menu-trigger">
      <slot name="trigger" :toggle="toggle" :isOpen="isOpen"></slot>
    </button>
    <transition
      enter-active-class="animate-slide-bottom-enter-active sm:animate-none sm:transition-all sm:duration-75 sm:ease-out-quad"
      leave-active-class="animate-slide-bottom-leave-active sm:animate-none sm:transition-all sm:duration-150 sm:ease-in-quad"
      enter-class="sm:opacity-0 sm:scale-75"
      enter-to-class="sm:opacity-100 sm:scale-100"
      leave-class="sm:opacity-100 sm:scale-100"
      leave-to-class="sm:opacity-0 sm:scale-75"
    >
      <div
        v-if="isOpen"
        class="fixed z-10 flex items-end overflow-hidden h-100 sm:absolute text-vanilla-200 transform-gpu bg-ink-400 bg-opacity-25 sm:bg-transparent sm:bg-opacity-0"
        :class="[directionClass, placementClasses]"
      >
        <div
          class="rounded-t-lg bg-ink-300 sm:rounded-sm shadow-double-dark"
          :class="`${sizeClass}`"
          ref="menu-body"
          v-outside-click="triggerClose"
        >
          <slot name="body" :isOpen="isOpen" :close="close"></slot>
        </div>
      </div>
    </transition>
  </span>
</template>

<script lang="ts">
import Vue from 'vue'
import { containsElement } from '@/helpers/components/utils'
import outsideClickDirective from '../../directives/outside-click'
Vue.directive('outside-click', outsideClickDirective)

export default Vue.extend({
  name: 'ZMenu',
  props: {
    direction: {
      type: String,
      default: 'right',
      validator: function(value: string): boolean {
        return ['left', 'right'].includes(value)
      }
    },
    placement: {
      type: String,
      default: 'bottom',
      validator: function(value: string): boolean {
        return ['top', 'bottom'].includes(value)
      }
    },
    size: {
      type: String,
      default: 'base',
      validator: function(value: string): boolean {
        return ['small', 'base', 'large'].includes(value)
      }
    },
    width: {
      type: String,
      default: 'base',
      validator: function(value: string): boolean {
        return ['x-small', 'small', 'base', 'large', 'x-large', '2x-large'].includes(value)
      }
    },
    collapseOnMobile: {
      type: Boolean,
      default: false
    }
  },
  data() {
    return {
      isOpen: false
    }
  },
  methods: {
    toggle(): void {
      this.isOpen = !this.isOpen
    },
    close(): void {
      this.isOpen = false
    },
    triggerClose(event?: Event): void {
      // Trigger only if open
      if (this.isOpen) {
        // If click event is not present close directly
        if (!event) {
          this.isOpen = false
        } else {
          event.stopImmediatePropagation()
          const target = event.target as HTMLElement
          const menuTrigger = this.$refs['menu-trigger'] as HTMLElement
          const menuBody = this.$refs['menu-body'] as HTMLElement

          if (!containsElement(menuTrigger, target) && !containsElement(menuBody, target)) {
            this.isOpen = false
          }
        }
      }
    }
  },
  computed: {
    directionClass(): string {
      return this.direction == 'right'
        ? 'sm:left-0 sm:origin-top-left'
        : 'sm:right-0 sm:origin-top-right'
    },
    placementClasses(): string {
      if (this.placement == 'top') {
        return 'inset-0 sm:inset-auto sm:bottom-10'
      }
      return 'inset-0 sm:inset-auto'
    },
    sizeClass(): string {
      const sizes: Record<string, string> = {
        small: `py-1 text-xs w-full ${this.widthClass} mt-5 sm:mt-2`,
        base: `py-1 text-sm w-full ${this.widthClass} mt-5 sm:mt-2`,
        large: `py-2 text-base w-full ${this.widthClass} mt-5 sm:mt-4`
      }
      return sizes[this.size || 'base']
    },
    widthClass(): string {
      const widths = {
        'x-small': 'sm:w-48',
        small: 'sm:w-52',
        base: 'sm:w-64',
        large: 'sm:w-72',
        'x-large': 'sm:w-80',
        '2x-large': 'sm:w-96'
      }
      return widths[this.width] || 'sm:w-64'
    }
  }
})
</script>
