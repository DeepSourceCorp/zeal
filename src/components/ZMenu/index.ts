import ZMenu from './ZMenu.vue'
import ZMenuItem from './ZMenuItem.vue'
import ZMenuSection from './ZMenuSection.vue'

export { ZMenu, ZMenuSection, ZMenuItem }

export default ZMenu
