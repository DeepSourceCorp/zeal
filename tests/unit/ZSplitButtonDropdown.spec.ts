import { mount } from '@vue/test-utils'

import ZButton from '../../src/components/ZButton'
import ZIcon from '../../src/components/ZIcon'
import ZMenu from '../../src/components/ZMenu'
import ZMenuItem from '../../src/components/ZMenu/ZMenuItem'
import ZSplitButtonDropdown from '../../src/components/ZSplitButtonDropdown'

describe('Split Button Dropdown component', () => {
  it('renders a top-level button and a menu with icon button as the trigger', () => {
    const wrapper = mount({
      template: `
      <z-split-button-dropdown :icon="icon" :label="label" :to="to">
        <z-menu size="small">
          <template v-slot:trigger="{ toggle }">
            <z-button button-type="secondary" size="small" @click="toggle">
              <z-icon icon="chevron-down" />
            </z-button>
          </template>

          <template #body>
            <z-menu-item as="button" icon="github">Create an issue on GitHub</z-menu-item>
            <z-menu-item as="button" icon="gitlab">Create an issue on GitLab</z-menu-item>
          </template>
        </z-menu>
    </z-split-button-dropdown>
    `,
      components: {
        ZButton,
        ZIcon,
        ZMenu,
        ZMenuItem,
        ZSplitButtonDropdown
      },
      data() {
        return {
          icon: 'github',
          label: 'GitHub',
          to: 'https://github.com'
        }
      }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('renders split button dropdown of various sizes', () => {
    const wrapper = mount({
      template: `
      <div>
        <z-split-button-dropdown v-for="size in sizes" :key="size" :icon="icon" :label="label" :to="to" :size="size">
          <z-menu size="small">
            <template v-slot:trigger="{ toggle }">
              <z-button button-type="secondary" :size="size" @click="toggle">
                <z-icon :size="iconSizes[size]" icon="chevron-down" />
              </z-button>
            </template>

            <template #body>
              <z-menu-item as="button" icon="github" @click="createIssue('github')">Create an issue on GitHub</z-menu-item>
              <z-menu-item as="button" icon="gitlab" @click="createIssue('gitlab')">Create an issue on GitLab</z-menu-item>
            </template>
          </z-menu>
      </z-split-button-dropdown>
    </div>
    `,
      components: {
        ZButton,
        ZIcon,
        ZMenu,
        ZMenuItem,
        ZSplitButtonDropdown
      },
      data() {
        return {
          icon: 'github',
          label: 'GitHub',
          to: 'https://github.com',
          sizes: ['small', 'medium', 'large', 'xlarge'],
          iconSizes: {
            small: 'small',
            medium: 'base',
            large: 'medium',
            xlarge: 'large'
          }
        }
      }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })
})
