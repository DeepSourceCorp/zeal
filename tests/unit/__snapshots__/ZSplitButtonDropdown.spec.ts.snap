// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Split Button Dropdown component renders a top-level button and a menu with icon button as the trigger 1`] = `
<div class="flex items-center"><a href="https://github.com" type="link" class="inline-flex items-center font-medium transition-colors duration-300 ease-in-out rounded-sm z-btn focus:outline-none whitespace-nowrap -mr-0.5 text-vanilla-100 justify-center p-0 h-8 px-4 py-1 text-xs space-x-1 leading-loose bg-ink-300 hover:bg-ink-200" rel="noopener noreferrer" target="_blank"><svg viewBox="0 0 24 24" stroke-linecap="round" stroke-linejoin="round" fill="none" class="z-icon w-4 h-4 z-icon--github text-vanilla-100 mr-1.5">
      <path d="M12 1C9.38799 1.00011 6.8613 1.92962 4.87196 3.62221C2.88263 5.31481 1.56048 7.66004 1.14208 10.2383C0.723674 12.8165 1.23633 15.4595 2.58831 17.6943C3.94029 19.9291 6.04337 21.61 8.52125 22.4361C9.07124 22.5324 9.27749 22.2024 9.27749 21.9136C9.27749 21.6524 9.26373 20.7861 9.26373 19.8649C6.49999 20.3736 5.785 19.1912 5.565 18.5724C5.32088 17.9706 4.93391 17.4373 4.43751 17.0186C4.05251 16.8124 3.50251 16.3036 4.42375 16.2899C4.77551 16.3281 5.11289 16.4505 5.40729 16.6468C5.7017 16.843 5.94446 17.1074 6.11499 17.4174C6.26544 17.6876 6.46774 17.9256 6.7103 18.1175C6.95286 18.3095 7.23091 18.4517 7.52852 18.536C7.82613 18.6203 8.13745 18.645 8.44464 18.6088C8.75184 18.5726 9.04886 18.4761 9.3187 18.3249C9.36632 17.7657 9.61554 17.2428 10.02 16.8537C7.57248 16.5787 5.015 15.6299 5.015 11.4225C4.99954 10.3292 5.40295 9.27151 6.14249 8.46623C5.80621 7.51608 5.84555 6.47336 6.25249 5.55125C6.25249 5.55125 7.1737 5.26249 9.27747 6.67874C11.0774 6.18372 12.9775 6.18372 14.7774 6.67874C16.8811 5.24875 17.8024 5.55125 17.8024 5.55125C18.2094 6.47334 18.2488 7.51609 17.9124 8.46623C18.6542 9.27013 19.0579 10.3288 19.0399 11.4225C19.0399 15.6436 16.4686 16.5787 14.0212 16.8537C14.2837 17.1197 14.4859 17.4392 14.614 17.7903C14.7421 18.1415 14.7932 18.516 14.7637 18.8887C14.7637 20.3599 14.7499 21.5424 14.7499 21.9136C14.7499 22.2024 14.9562 22.5461 15.5062 22.4361C17.9797 21.6033 20.0767 19.9185 21.4229 17.6825C22.7692 15.4466 23.2769 12.8049 22.8556 10.2292C22.4343 7.65346 21.1114 5.31128 19.1229 3.62074C17.1345 1.93021 14.6099 1.00136 12 1V1Z" fill="currentColor"></path>
    </svg>
    GitHub
  </a>
  <div class="relative z-divider inline-block w-px h-4 mx-0 align-middle bg-slate">
    <!---->
  </div> <span class="relative z-menu"><div class="leading-none"><button href="#" type="button" class="inline-flex items-center font-medium transition-colors duration-300 ease-in-out rounded-sm z-btn focus:outline-none whitespace-nowrap text-vanilla-100 justify-center h-8 px-4 py-1 text-xs space-x-1 leading-loose bg-ink-300 hover:bg-ink-200"><!----> <svg viewBox="0 0 24 24" stroke-linecap="round" stroke-linejoin="round" fill="none" class="z-icon w-4 h-4 z-icon--chevron-down text-vanilla-400"><polyline points="6 9 12 15 18 9"></polyline></svg></button></div> <transition-stub enterclass="sm:opacity-0 sm:scale-75" leaveclass="sm:opacity-100 sm:scale-100" entertoclass="sm:opacity-100 sm:scale-100" leavetoclass="sm:opacity-0 sm:scale-75" enteractiveclass="animate-slide-bottom-enter-active sm:animate-none sm:transition-all sm:duration-75 sm:ease-out-quad" leaveactiveclass="animate-slide-bottom-leave-active sm:animate-none sm:transition-all sm:duration-150 sm:ease-in-quad"><!----></transition-stub></span>
</div>
`;

exports[`Split Button Dropdown component renders split button dropdown of various sizes 1`] = `
<div>
  <div class="flex items-center"><a href="https://github.com" type="link" class="inline-flex items-center font-medium transition-colors duration-300 ease-in-out rounded-sm z-btn focus:outline-none whitespace-nowrap -mr-0.5 text-vanilla-100 justify-center p-0 h-8 px-4 py-1 text-xs space-x-1 leading-loose bg-ink-300 hover:bg-ink-200" rel="noopener noreferrer" target="_blank"><svg viewBox="0 0 24 24" stroke-linecap="round" stroke-linejoin="round" fill="none" class="z-icon w-4 h-4 z-icon--github text-vanilla-100 mr-1.5">
        <path d="M12 1C9.38799 1.00011 6.8613 1.92962 4.87196 3.62221C2.88263 5.31481 1.56048 7.66004 1.14208 10.2383C0.723674 12.8165 1.23633 15.4595 2.58831 17.6943C3.94029 19.9291 6.04337 21.61 8.52125 22.4361C9.07124 22.5324 9.27749 22.2024 9.27749 21.9136C9.27749 21.6524 9.26373 20.7861 9.26373 19.8649C6.49999 20.3736 5.785 19.1912 5.565 18.5724C5.32088 17.9706 4.93391 17.4373 4.43751 17.0186C4.05251 16.8124 3.50251 16.3036 4.42375 16.2899C4.77551 16.3281 5.11289 16.4505 5.40729 16.6468C5.7017 16.843 5.94446 17.1074 6.11499 17.4174C6.26544 17.6876 6.46774 17.9256 6.7103 18.1175C6.95286 18.3095 7.23091 18.4517 7.52852 18.536C7.82613 18.6203 8.13745 18.645 8.44464 18.6088C8.75184 18.5726 9.04886 18.4761 9.3187 18.3249C9.36632 17.7657 9.61554 17.2428 10.02 16.8537C7.57248 16.5787 5.015 15.6299 5.015 11.4225C4.99954 10.3292 5.40295 9.27151 6.14249 8.46623C5.80621 7.51608 5.84555 6.47336 6.25249 5.55125C6.25249 5.55125 7.1737 5.26249 9.27747 6.67874C11.0774 6.18372 12.9775 6.18372 14.7774 6.67874C16.8811 5.24875 17.8024 5.55125 17.8024 5.55125C18.2094 6.47334 18.2488 7.51609 17.9124 8.46623C18.6542 9.27013 19.0579 10.3288 19.0399 11.4225C19.0399 15.6436 16.4686 16.5787 14.0212 16.8537C14.2837 17.1197 14.4859 17.4392 14.614 17.7903C14.7421 18.1415 14.7932 18.516 14.7637 18.8887C14.7637 20.3599 14.7499 21.5424 14.7499 21.9136C14.7499 22.2024 14.9562 22.5461 15.5062 22.4361C17.9797 21.6033 20.0767 19.9185 21.4229 17.6825C22.7692 15.4466 23.2769 12.8049 22.8556 10.2292C22.4343 7.65346 21.1114 5.31128 19.1229 3.62074C17.1345 1.93021 14.6099 1.00136 12 1V1Z" fill="currentColor"></path>
      </svg>
      GitHub
    </a>
    <div class="relative z-divider inline-block w-px h-4 mx-0 align-middle bg-slate">
      <!---->
    </div> <span class="relative z-menu"><div class="leading-none"><button href="#" type="button" class="inline-flex items-center font-medium transition-colors duration-300 ease-in-out rounded-sm z-btn focus:outline-none whitespace-nowrap text-vanilla-100 justify-center h-8 px-4 py-1 text-xs space-x-1 leading-loose bg-ink-300 hover:bg-ink-200"><!----> <svg viewBox="0 0 24 24" stroke-linecap="round" stroke-linejoin="round" fill="none" class="z-icon w-4 h-4 z-icon--chevron-down text-vanilla-400"><polyline points="6 9 12 15 18 9"></polyline></svg></button></div> <transition-stub enterclass="sm:opacity-0 sm:scale-75" leaveclass="sm:opacity-100 sm:scale-100" entertoclass="sm:opacity-100 sm:scale-100" leavetoclass="sm:opacity-0 sm:scale-75" enteractiveclass="animate-slide-bottom-enter-active sm:animate-none sm:transition-all sm:duration-75 sm:ease-out-quad" leaveactiveclass="animate-slide-bottom-leave-active sm:animate-none sm:transition-all sm:duration-150 sm:ease-in-quad"><!----></transition-stub></span>
  </div>
  <div class="flex items-center"><a href="https://github.com" type="link" class="inline-flex items-center font-medium transition-colors duration-300 ease-in-out rounded-sm z-btn focus:outline-none whitespace-nowrap -mr-0.5 text-vanilla-100 justify-center p-0 h-10 px-4 py-2 text-base space-x-2 leading-8 bg-ink-300 hover:bg-ink-200" rel="noopener noreferrer" target="_blank"><svg viewBox="0 0 24 24" stroke-linecap="round" stroke-linejoin="round" fill="none" class="z-icon w-5 h-5 z-icon--github text-vanilla-100 mr-2">
        <path d="M12 1C9.38799 1.00011 6.8613 1.92962 4.87196 3.62221C2.88263 5.31481 1.56048 7.66004 1.14208 10.2383C0.723674 12.8165 1.23633 15.4595 2.58831 17.6943C3.94029 19.9291 6.04337 21.61 8.52125 22.4361C9.07124 22.5324 9.27749 22.2024 9.27749 21.9136C9.27749 21.6524 9.26373 20.7861 9.26373 19.8649C6.49999 20.3736 5.785 19.1912 5.565 18.5724C5.32088 17.9706 4.93391 17.4373 4.43751 17.0186C4.05251 16.8124 3.50251 16.3036 4.42375 16.2899C4.77551 16.3281 5.11289 16.4505 5.40729 16.6468C5.7017 16.843 5.94446 17.1074 6.11499 17.4174C6.26544 17.6876 6.46774 17.9256 6.7103 18.1175C6.95286 18.3095 7.23091 18.4517 7.52852 18.536C7.82613 18.6203 8.13745 18.645 8.44464 18.6088C8.75184 18.5726 9.04886 18.4761 9.3187 18.3249C9.36632 17.7657 9.61554 17.2428 10.02 16.8537C7.57248 16.5787 5.015 15.6299 5.015 11.4225C4.99954 10.3292 5.40295 9.27151 6.14249 8.46623C5.80621 7.51608 5.84555 6.47336 6.25249 5.55125C6.25249 5.55125 7.1737 5.26249 9.27747 6.67874C11.0774 6.18372 12.9775 6.18372 14.7774 6.67874C16.8811 5.24875 17.8024 5.55125 17.8024 5.55125C18.2094 6.47334 18.2488 7.51609 17.9124 8.46623C18.6542 9.27013 19.0579 10.3288 19.0399 11.4225C19.0399 15.6436 16.4686 16.5787 14.0212 16.8537C14.2837 17.1197 14.4859 17.4392 14.614 17.7903C14.7421 18.1415 14.7932 18.516 14.7637 18.8887C14.7637 20.3599 14.7499 21.5424 14.7499 21.9136C14.7499 22.2024 14.9562 22.5461 15.5062 22.4361C17.9797 21.6033 20.0767 19.9185 21.4229 17.6825C22.7692 15.4466 23.2769 12.8049 22.8556 10.2292C22.4343 7.65346 21.1114 5.31128 19.1229 3.62074C17.1345 1.93021 14.6099 1.00136 12 1V1Z" fill="currentColor"></path>
      </svg>
      GitHub
    </a>
    <div class="relative z-divider inline-block w-px h-4 mx-0 align-middle bg-slate h-6">
      <!---->
    </div> <span class="relative z-menu"><div class="leading-none"><button href="#" type="button" class="inline-flex items-center font-medium transition-colors duration-300 ease-in-out rounded-sm z-btn focus:outline-none whitespace-nowrap text-vanilla-100 justify-center h-10 px-4 py-2 text-base space-x-2 leading-8 bg-ink-300 hover:bg-ink-200"><!----> <svg viewBox="0 0 24 24" stroke-linecap="round" stroke-linejoin="round" fill="none" class="z-icon w-5 h-5 z-icon--chevron-down text-vanilla-400"><polyline points="6 9 12 15 18 9"></polyline></svg></button></div> <transition-stub enterclass="sm:opacity-0 sm:scale-75" leaveclass="sm:opacity-100 sm:scale-100" entertoclass="sm:opacity-100 sm:scale-100" leavetoclass="sm:opacity-0 sm:scale-75" enteractiveclass="animate-slide-bottom-enter-active sm:animate-none sm:transition-all sm:duration-75 sm:ease-out-quad" leaveactiveclass="animate-slide-bottom-leave-active sm:animate-none sm:transition-all sm:duration-150 sm:ease-in-quad"><!----></transition-stub></span>
  </div>
  <div class="flex items-center"><a href="https://github.com" type="link" class="inline-flex items-center font-medium transition-colors duration-300 ease-in-out rounded-sm z-btn focus:outline-none whitespace-nowrap -mr-0.5 text-vanilla-100 justify-center p-0 h-12 px-4 py-2.5 text-lg space-x-2.5 leading-9 bg-ink-300 hover:bg-ink-200" rel="noopener noreferrer" target="_blank"><svg viewBox="0 0 24 24" stroke-linecap="round" stroke-linejoin="round" fill="none" class="z-icon w-6 h-6 z-icon--github text-vanilla-100 mr-2.5">
        <path d="M12 1C9.38799 1.00011 6.8613 1.92962 4.87196 3.62221C2.88263 5.31481 1.56048 7.66004 1.14208 10.2383C0.723674 12.8165 1.23633 15.4595 2.58831 17.6943C3.94029 19.9291 6.04337 21.61 8.52125 22.4361C9.07124 22.5324 9.27749 22.2024 9.27749 21.9136C9.27749 21.6524 9.26373 20.7861 9.26373 19.8649C6.49999 20.3736 5.785 19.1912 5.565 18.5724C5.32088 17.9706 4.93391 17.4373 4.43751 17.0186C4.05251 16.8124 3.50251 16.3036 4.42375 16.2899C4.77551 16.3281 5.11289 16.4505 5.40729 16.6468C5.7017 16.843 5.94446 17.1074 6.11499 17.4174C6.26544 17.6876 6.46774 17.9256 6.7103 18.1175C6.95286 18.3095 7.23091 18.4517 7.52852 18.536C7.82613 18.6203 8.13745 18.645 8.44464 18.6088C8.75184 18.5726 9.04886 18.4761 9.3187 18.3249C9.36632 17.7657 9.61554 17.2428 10.02 16.8537C7.57248 16.5787 5.015 15.6299 5.015 11.4225C4.99954 10.3292 5.40295 9.27151 6.14249 8.46623C5.80621 7.51608 5.84555 6.47336 6.25249 5.55125C6.25249 5.55125 7.1737 5.26249 9.27747 6.67874C11.0774 6.18372 12.9775 6.18372 14.7774 6.67874C16.8811 5.24875 17.8024 5.55125 17.8024 5.55125C18.2094 6.47334 18.2488 7.51609 17.9124 8.46623C18.6542 9.27013 19.0579 10.3288 19.0399 11.4225C19.0399 15.6436 16.4686 16.5787 14.0212 16.8537C14.2837 17.1197 14.4859 17.4392 14.614 17.7903C14.7421 18.1415 14.7932 18.516 14.7637 18.8887C14.7637 20.3599 14.7499 21.5424 14.7499 21.9136C14.7499 22.2024 14.9562 22.5461 15.5062 22.4361C17.9797 21.6033 20.0767 19.9185 21.4229 17.6825C22.7692 15.4466 23.2769 12.8049 22.8556 10.2292C22.4343 7.65346 21.1114 5.31128 19.1229 3.62074C17.1345 1.93021 14.6099 1.00136 12 1V1Z" fill="currentColor"></path>
      </svg>
      GitHub
    </a>
    <div class="relative z-divider inline-block w-px h-4 mx-0 align-middle bg-slate h-8">
      <!---->
    </div> <span class="relative z-menu"><div class="leading-none"><button href="#" type="button" class="inline-flex items-center font-medium transition-colors duration-300 ease-in-out rounded-sm z-btn focus:outline-none whitespace-nowrap text-vanilla-100 justify-center h-12 px-4 py-2.5 text-lg space-x-2.5 leading-9 bg-ink-300 hover:bg-ink-200"><!----> <svg viewBox="0 0 24 24" stroke-linecap="round" stroke-linejoin="round" fill="none" class="z-icon w-6 h-6 z-icon--chevron-down text-vanilla-400"><polyline points="6 9 12 15 18 9"></polyline></svg></button></div> <transition-stub enterclass="sm:opacity-0 sm:scale-75" leaveclass="sm:opacity-100 sm:scale-100" entertoclass="sm:opacity-100 sm:scale-100" leavetoclass="sm:opacity-0 sm:scale-75" enteractiveclass="animate-slide-bottom-enter-active sm:animate-none sm:transition-all sm:duration-75 sm:ease-out-quad" leaveactiveclass="animate-slide-bottom-leave-active sm:animate-none sm:transition-all sm:duration-150 sm:ease-in-quad"><!----></transition-stub></span>
  </div>
  <div class="flex items-center"><a href="https://github.com" type="link" class="inline-flex items-center font-medium transition-colors duration-300 ease-in-out rounded-sm z-btn focus:outline-none whitespace-nowrap -mr-0.5 text-vanilla-100 justify-center p-0 h-13 px-4 py-3 text-lg space-x-3 leading-9 bg-ink-300 hover:bg-ink-200" rel="noopener noreferrer" target="_blank"><svg viewBox="0 0 24 24" stroke-linecap="round" stroke-linejoin="round" fill="none" class="z-icon w-8 h-8 z-icon--github text-vanilla-100 mr-3">
        <path d="M12 1C9.38799 1.00011 6.8613 1.92962 4.87196 3.62221C2.88263 5.31481 1.56048 7.66004 1.14208 10.2383C0.723674 12.8165 1.23633 15.4595 2.58831 17.6943C3.94029 19.9291 6.04337 21.61 8.52125 22.4361C9.07124 22.5324 9.27749 22.2024 9.27749 21.9136C9.27749 21.6524 9.26373 20.7861 9.26373 19.8649C6.49999 20.3736 5.785 19.1912 5.565 18.5724C5.32088 17.9706 4.93391 17.4373 4.43751 17.0186C4.05251 16.8124 3.50251 16.3036 4.42375 16.2899C4.77551 16.3281 5.11289 16.4505 5.40729 16.6468C5.7017 16.843 5.94446 17.1074 6.11499 17.4174C6.26544 17.6876 6.46774 17.9256 6.7103 18.1175C6.95286 18.3095 7.23091 18.4517 7.52852 18.536C7.82613 18.6203 8.13745 18.645 8.44464 18.6088C8.75184 18.5726 9.04886 18.4761 9.3187 18.3249C9.36632 17.7657 9.61554 17.2428 10.02 16.8537C7.57248 16.5787 5.015 15.6299 5.015 11.4225C4.99954 10.3292 5.40295 9.27151 6.14249 8.46623C5.80621 7.51608 5.84555 6.47336 6.25249 5.55125C6.25249 5.55125 7.1737 5.26249 9.27747 6.67874C11.0774 6.18372 12.9775 6.18372 14.7774 6.67874C16.8811 5.24875 17.8024 5.55125 17.8024 5.55125C18.2094 6.47334 18.2488 7.51609 17.9124 8.46623C18.6542 9.27013 19.0579 10.3288 19.0399 11.4225C19.0399 15.6436 16.4686 16.5787 14.0212 16.8537C14.2837 17.1197 14.4859 17.4392 14.614 17.7903C14.7421 18.1415 14.7932 18.516 14.7637 18.8887C14.7637 20.3599 14.7499 21.5424 14.7499 21.9136C14.7499 22.2024 14.9562 22.5461 15.5062 22.4361C17.9797 21.6033 20.0767 19.9185 21.4229 17.6825C22.7692 15.4466 23.2769 12.8049 22.8556 10.2292C22.4343 7.65346 21.1114 5.31128 19.1229 3.62074C17.1345 1.93021 14.6099 1.00136 12 1V1Z" fill="currentColor"></path>
      </svg>
      GitHub
    </a>
    <div class="relative z-divider inline-block w-px h-4 mx-0 align-middle bg-slate h-10">
      <!---->
    </div> <span class="relative z-menu"><div class="leading-none"><button href="#" type="button" class="inline-flex items-center font-medium transition-colors duration-300 ease-in-out rounded-sm z-btn focus:outline-none whitespace-nowrap text-vanilla-100 justify-center h-13 px-4 py-3 text-lg space-x-3 leading-9 bg-ink-300 hover:bg-ink-200"><!----> <svg viewBox="0 0 24 24" stroke-linecap="round" stroke-linejoin="round" fill="none" class="z-icon w-8 h-8 z-icon--chevron-down text-vanilla-400"><polyline points="6 9 12 15 18 9"></polyline></svg></button></div> <transition-stub enterclass="sm:opacity-0 sm:scale-75" leaveclass="sm:opacity-100 sm:scale-100" entertoclass="sm:opacity-100 sm:scale-100" leavetoclass="sm:opacity-0 sm:scale-75" enteractiveclass="animate-slide-bottom-enter-active sm:animate-none sm:transition-all sm:duration-75 sm:ease-out-quad" leaveactiveclass="animate-slide-bottom-leave-active sm:animate-none sm:transition-all sm:duration-150 sm:ease-in-quad"><!----></transition-stub></span>
  </div>
</div>
`;
